Hosting API Documentation.md

```diff
@@ -0,0 +1,112 @@
+
+- Write tests using rspec and annotate them using [dox](https://github.com/infinum/dox)
+- Once tests on [semaphore](https://semaphoreci.com) pass we move to deploy
+- In the deploy phase we do the following:
+  * deploy the application using [mina](https://github.com/mina-deploy/mina)
+  * run only dox tests to generate the documentation
+  * copy the generated documentation to the application server inside `public` folder
+
+# Why
+
+We settled on serving our documentation for the application server because it is convenient, easiest
+and free.
+
+Solutions like apiary.io require us to have one project for each environemnt, can't easily set
+a desired domain and can't easily hide behind some kind of authentication (like Basic Authentication).
+
+Copying after a successful deploy we assure that the documentation is always up to date.
+
+Serving from the `public` folder we are not going through the rails stack but we let nginx serve
+the static file.
+
+# How
+
+## Dox generating task
+
+Here is the advanced example of a rake task that generates the documentation:
+It has the feature to write ERB code inside your acompanying .md files.
+
+```ruby
+namespace :dox do
+  task :md, [:version, :docs_path, :host] do |_, args|
+    require 'rspec/core/rake_task'
+
+    version = args[:version] || :v1
+    docs_path = args[:docs_path] || "api/#{version}/docs"
+
+    RSpec::Core::RakeTask.new(:api_spec) do |t|
+      t.pattern = "spec/requests/#{version}/"
+      t.rspec_opts =
+        "-f Dox::Formatter --order defined -t dox -o public/#{docs_path}/apispec.md.erb"
+    end
+
+    Rake::Task['api_spec'].invoke
+  end
+
+  task :html, [:version, :docs_path, :host] => :erb do |_, args|
+    version = args[:version] || :v1
+    docs_path = args[:docs_path] || "api/#{version}/docs"
+    `node_modules/.bin/aglio -i public/#{docs_path}/apispec.md -o public/#{docs_path}/index.html --theme-full-width --theme-variables flatly`
+  end
+
+  task :open, [:version, :docs_path, :host] => :html do |_, args|
+    version = args[:version] || :v1
+    docs_path = args[:docs_path] || "api/#{version}/docs"
+
+    `open public/#{docs_path}/index.html`
+  end
+
+  task :erb, [:version, :docs_path, :host] => [:environment, :md] do |_, args|
+    require 'erb'
+
+    host = args[:host] || 'http://localhost:3000'
+    version = args[:version] || :v1
+    docs_path = args[:docs_path] || "api/#{version}/docs"
+
+    File.open("public/#{docs_path}/apispec.md", 'w') do |f|
+      f.write(ERB.new(File.read("public/#{docs_path}/apispec.md.erb")).result(binding))
+    end
+  end
+end
+```
+
+```markdown
+FORMAT: 1A
+HOST: <%= host %>
+
+
+# My app API
+An API documentation for my app
+```
+
+```sh
+rake 'api:doc:html[v1, api/v1/docs, https://example.com]'
+```
+
+The above example can be simplified if some of the features are not needed.
+
+## Deploy script
+
+We are using [mina-dox](https://github.com/infinum/mina-dox) to get `dox:publish` mina task.
```

Comment:
The mina-dox extension exposes the `dox:publich` task, which runs `rake dox:html` and scps the generated documentation.

Reply:
`dox:publish` :)